core/src/main/java/com/squarespace/template/plugins/platform/ContentFormatters.java

```diff
@@ -320,6 +323,90 @@ public void apply(Context ctx, Arguments args, Variables variables) throws CodeE
     }
   };
 
+  public static class WebsiteColorFormatter extends BaseFormatter {
+
+    public WebsiteColorFormatter() {
+      super("website-color", false);
+    }
+
+    private final DecimalFormat format = new DecimalFormat("0.#");
+
+    @Override
+    public void apply(Context ctx, Arguments args, Variables variables) throws CodeExecuteException {
+      Variable var = variables.first();
+      JsonNode node = var.node();
+      boolean hasAlphaValue = false;
+      String errorMessage = "";
+
+      JsonNode hueNode = node.path("hue");
+      JsonNode saturationNode = node.path("saturation");
+      JsonNode lightnessNode = node.path("lightness");
+      if (hueNode.isMissingNode() || saturationNode.isMissingNode() || lightnessNode.isMissingNode()) {
+        errorMessage = "Missing an H/S/L value.";
+        var.set(errorMessage);
+        return;
+      }
+
+      double hue = hueNode.asDouble();
+      double saturation = saturationNode.asDouble();
+      double lightness = lightnessNode.asDouble();
+      JsonNode alphaNode = node.path("alpha");
+      Double alpha = null;
+      if (!alphaNode.isMissingNode()) {
+        hasAlphaValue = true;
+        alpha = alphaNode.asDouble();
+      }
+
+      if (hue < 0 || hue > 360) {
+        errorMessage += "Hue out of bounds. ";
+      }
+
+      if (saturation >= 0 && saturation <= 1) {
+        saturation *= 100;
+      } else {
+        errorMessage += "Saturation out of bounds. ";
```

Comment:
nit: should these additions to the error message include `\n` characters as well? Or is it preferred to have them as a single line

Reply:
we want to keep the error messages as concise as people since longer strings will increase the code size for the Typescript compiler